### PR TITLE
Bjorncs/tls certificate validation

### PR DESCRIPTION
**Pull request by bjorncs**

### CODE REVIEW

File: security-utils/src/main/java/com/yahoo/security/tls/DefaultTlsContext.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    private static void restrictSetOfEnabledCiphers(SSLEngine sslEngine) {
+        String[] validCipherSuits = Arrays.stream(sslEngine.getSupportedCipherSuites())
+                .filter(ALLOWED_CIPHER_SUITS::contains)
+                .toArray(String[]::new);
+        if (validCipherSuits.length == 0) {
+            throw new IllegalStateException("None of the allowed cipher suits are supported");
+        }
+        log.log(Level.FINE, () -> String.format("Allowed cipher suits that are supported: %s", Arrays.toString(validCipherSuits)));
+        sslEngine.setEnabledCipherSuites(validCipherSuits);
+    }
```
Comment (vekterli) [inline security-utils/src/main/java/com/yahoo/security/tls/DefaultTlsContext.java]: s/suits/suites/ in this method

File: security-utils/src/main/java/com/yahoo/security/tls/DefaultTlsContext.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+            "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+            "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256",
+            "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256",
+            "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+            "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+            "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384",
+            "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384",
+            "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
+            "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256");
```
Comment (vekterli) [inline security-utils/src/main/java/com/yahoo/security/tls/DefaultTlsContext.java]: Do we want to support AES CBC codecs at all or just GCM?

**Review by vekterli — APPROVED** [review]
:+1: just a couple of minor comments and a question

File: security-utils/src/main/java/com/yahoo/security/tls/DefaultTlsContext.java (hunk 2 above)
Comment (bjorncs) [reply security-utils/src/main/java/com/yahoo/security/tls/DefaultTlsContext.java]: Only GCM ciphers is fine - JDK8 supports the relevant GCM ciphers.

**Review by vekterli — APPROVED** [review]

**Pull request merged**